storage/s3.go

```diff
@@ -126,13 +126,80 @@ func (s *S3) Get(ctx context.Context, path string) (io.ReadCloser, error) {
 		Bucket: &s.cfg.Bucket,
 		Key:    &path,
 	}
+
 	out, err := s.s3client.GetObject(ctx, input)
 	if err != nil {
 		return nil, errors.Wrapf(err, "fail to get S3 object %v", path)
 	}
 	return out.Body, nil
 }
 
+// GetWithRetries function downloads the object from S3 and writes it to the
+// writer. It returns the number of bytes written to the writer.
+//
+// It implements a retry mechanism to handle if connection is closed before
+// the end of the download.
```

Comment:
From what I understand from the updated PR title, this comment is no longer accurate
```suggestion
// It implements a retry mechanism to handle if connection is closed before
// the end of the download or if all data couldn't be read.
```